recipes/util-linux-libuuid/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/util-linux/util-linux.git"
+    license = "BSD-3-Clause"
+    topics = "id", "identifier", "unique", "uuid"
+    package_type = "library"
+    provides = "libuuid"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _has_sys_file_header(self):
+        return self.settings.os in ["FreeBSD", "Linux", "Macos"]
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def _minimum_compiler_version(self, compiler, build_type):
+        min_version = {
+            "gcc": {
+                "Release": "4",
+                "Debug": "8",
+            },
+            "clang": {
+                "Release": "3",
+                "Debug": "3",
+            },
+            "apple-clang": {
+                "Release": "5",
+                "Debug": "5",
+            },
+        }
+        return min_version.get(str(compiler), {}).get(str(build_type), "0")
+
+    def validate(self):
+        min_version = self._minimum_compiler_version(self.settings.compiler, self.settings.build_type)
+        if Version(self.settings.compiler.version) < min_version:
+            raise ConanInvalidConfiguration(f"{self.settings.compiler} {self.settings.compiler.version} does not meet the minimum version requirement of version {min_version}")
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(f"{self.ref} is not supported on Windows")
+        if self.settings.os == "Macos":
+            # FIXME: Add Macos compatibility. This is currently breaking because builds are unable to find libtool-2
+            # This is a bit puzzling given `libtool` is a tool_requires, and I haven't been able to replicate this error
+            # locally.
+            raise ConanInvalidConfiguration(f"{self.ref} is not currently supported on Macos. Please contribute this functionality if you require it.")
+
+    def requirements(self):
+        if self.settings.os == "Macos":
+            # Required because libintl.{a,dylib} is not distributed via libc on Macos
+            self.requires("libgettext/0.21")
+
+    def build_requirements(self):
+        self.tool_requires("libtool/2.4.7")
+        self.tool_requires("m4/1.4.19")
+        self.tool_requires("pkgconf/1.9.3")
+        self.tool_requires("bison/3.8.2")
+        self.tool_requires("autoconf/2.71")
+        self.tool_requires("automake/1.16.5")
```

Comment:
Very bad idea to explicitly reference automake, autoconf and m4, it doesn't scale in build requirements, it's well known. Why this recipe doesn't follow other autoconf recipes guidelines?

Reply:
I'm not experienced with automake. I used a few other recipes as guidelines but they obviously haven't incorporated the practices you're talking about and it didn't come up in review. It would be useful if those were documented somewhere easily discoverable. Feel free to submit a new PR with the necessary changes if that's a better way of managing these.

Reply:
A good reference is autotools recipes template.

(sorry, I won't fix this, I don't have the time. It will break when someone will bump automake version in libtool recipe)

Reply:
Good to know, but it wouldn't have addressed this - it has an explicit example requiring automake, and no discussion of when it shouldn't be required.

Reply:
> it has an explicit example requiring automake, and no discussion of when it shouldn't be required.

There is a comment explaining when it's required: https://github.com/conan-io/conan-center-index/blob/60fa77f44eafacf99b7f1d43650f13dd8db5d971/docs/package_templates/autotools_package/all/conanfile.py#L93C1-L96